modules/react/breadcrumbs/lib/BreadcrumbsCurrentItem.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import {CSSObject} from '@emotion/styled';
+import {
+  styled,
+  createElemPropsHook,
+  useLocalRef,
+  composeHooks,
+  createSubcomponent,
+} from '@workday/canvas-kit-react/common';
+import {type} from '@workday/canvas-kit-react/tokens';
+import {TooltipContainer} from '@workday/canvas-kit-react/tooltip';
+import {Popper} from '@workday/canvas-kit-react/popup';
+import {
+  useListItemRegister,
+  useOverflowListItemMeasure,
+} from '@workday/canvas-kit-react/collection';
+import {useTruncateTooltip} from './hooks/useTruncateTooltip';
+import {useBreadcrumbsModel} from './hooks/useBreadcrumbsModel';
+
+// default max-width for truncating text
+const DEFAULT_MAX_WIDTH = 350;
+
+export const truncateStyles = (maxWidth: number = DEFAULT_MAX_WIDTH): CSSObject => ({
+  display: 'inline-block',
+  maxWidth,
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap',
+  minWidth: '94px',
+});
+
+export interface BreadcrumbsCurrentItemProps extends React.HTMLAttributes<HTMLLIElement> {
+  /**
+   * The max-width of the text
+   *
+   * @default 350px
+   */
+  maxWidth?: number;
+}
+
+const ListItem = styled('li')(
+  {
+    display: 'flex',
+    alignItems: 'center',
+    flexDirection: 'column',
+    textAlign: 'left',
+    ...type.levels.subtext.large,
```

Comment:
can you use your text component styles here?

Reply:
`<Text as='li'..../>`

Reply:
I changed this part. Unfortunately, I faced with issue with using Text and Hyperlink components in `Breadcrumbs.Link`. If I pass `Hyperlink` as `as` prop it passes all text style props as attributes. So I decided to keep `Hyperlink` for `Breadcrumbs.Link` for now. I guess we can review if `Hyperlink` component should be built based on `Text` component.